example/App.js

```diff
@@ -53,6 +53,43 @@ export default class App extends Component {
     }
   };
 
+  captureScreenshot = () =>
+    captureScreen(this.state.value)
+      .then(
+        res =>
+          this.state.value.result !== "tmpfile"
+          ? res
+          : new Promise((success, failure) =>
+          // just a test to ensure res can be used in Image.getSize
+          Image.getSize(
+            res,
+            (width, height) => (
+              console.log(res, width, height), success(res)
+            ),
+            failure
+          )
+        )
+      )
+      .then(res =>
+        this.setState({
+          error: null,
+          res,
+          previewSource: {
+            uri:
+              this.state.value.result === "base64"
+                ? "data:image/" + this.state.value.format + ";base64," + res
+                : res
+          }
+        })
+      )
+      .catch(
+        error => (
+          console.warn(error),
+          this.setState({ error, res: null, previewSource: null })
+        )
+      );
+    
+
   snapshot = refname => () =>
     captureRef(this.refs[refname], this.state.value)
```

Comment:
little hack suggestion: could you actually reuse the snapshot method and we would just consider refname, if falsy, means it capture the screen?

something like:

```js 
    (refname ? captureRef(this.refs[refname], this.state.value) : captureScreen(this.state.value))
  .then(...
```

and would just call

```
onPress={this.snapshot()}
```